device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -469,10 +471,16 @@ def _init_cmis4_module(self, port_num, xcvr, hwsku):
         self._write_byte(port_num, self.IDENTITY_EEPROM_ADDR, 0x10, 143, 0xff)
         # Initialize datapath
         self._write_byte(port_num, self.IDENTITY_EEPROM_ADDR, 0x10, 128, 0x00)
-        time.sleep(1)
+
         # Validate configuration status
-        buf = self._read_eeprom_devid(port_num, self.IDENTITY_EEPROM_ADDR, self._page_to_flat(202, 0x11), 4)
-        err = "".join(buf)
+        err = '00000000'
+        for t in range(10):
```

Comment:
Eoptolink CMIS4 took 6 seconds for config status to get populated to the register, and hence 10 sec should be sufficient